meta-luneos/conf/distro/include/luneos.inc

```diff
@@ -140,7 +140,7 @@ ERROR_QA_remove = " version-going-backwards"
 
 # Default WEBOS_TARGET_* values (see webos_cmake.bbclass)
 WEBOS_TARGET_CORE_OS ?= "rockhopper"
-WEBOS_TARGET_MACHINE_IMPL ?= "device"
+WEBOS_TARGET_MACHINE_IMPL ?= "hardware"
 # For qemu we build for the emulator but for all other targets for a real device
 WEBOS_TARGET_MACHINE_IMPL_qemuall = "emulator"
```

Comment:
What happends if you remove all it all and leave it only in webos_machine_impl_dep?

Reply:
When I remove it, it will still show "device" in my /etc/palm/luna-platform.conf which is incorrect. 

OSE does this: https://github.com/webosose/meta-webosose/blob/e50f1d1b5350491a71c9d0908a3ef725596c2766/meta-webos-raspberrypi/conf/machine/include/webos-rpi.inc#L7

Seeing our default is hardware and only qemu is emulator, I would suggest to merge this as is and we override it anyway already for qemu with https://github.com/webOS-ports/meta-webos-ports/blob/warrior/meta-luneos/conf/distro/include/luneos.inc#L153

Otherwise we would need to specify it each time in our machine.conf for each device, which doesn't make a lot of sense to me.

Reply:
For which MACHINE does it show "device" and shouldn't?

And do you mean Type=device? That seems to be explicitly set for most MACHINEs in:

meta-webos-ports $ grep Type meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/*/*
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/a500/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/athene/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/grouper/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/hammerhead/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/maguro/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/mako/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/mido/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/onyx/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/pinephone/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/qemux86-64/luna-platform.conf:Type=emulator
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/qemux86/luna-platform.conf:Type=emulator
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/raspberrypi2/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/raspberrypi3/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/rosy/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/tenderloin/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/tissot/luna-platform.conf:Type=device
meta-luneos/recipes-webos/luna-sysmgr-conf/luna-sysmgr-conf/tuna/luna-platform.conf:Type=device

and luna-sysmgr-conf shouldn't be using WEBOS_MACHINE_IMPL_DEP as it doesn't inherit the bbclass mentioned before (so the "desktop" directory there doesn't make much sense).

Reply:
It should show "hardware" everywhere it shows "device" ;) I could update it there as well of course. Just wonder what would be the best approach going forward.

Reply:
I don't think so, Type in luna-sysmgr isn't the same as WEBOS_TARGET_MACHINE_IMPL, see:

https://github.com/webOS-ports/luna-sysmgr-common/blob/webOS-ports/master/Src/base/settings/Settings.cpp#L58
and
https://github.com/webOS-ports/luna-sysmgr-common/blob/webOS-ports/master/Src/base/settings/Settings.cpp#L612